P2PEnabledConference.js

```diff
@@ -0,0 +1,947 @@
+                        logger.error(
+                            'Failed to accept incoming P2P Jingle session',
+                            error);
+                    }
+                );
+            },
+            error => {
+                logger.error(
+                    `Failed to add ${localTracks} to the P2P connection`,
+                    error);
+            });
+    }
+
+    /**
+     * @inheritDoc
+     * @override
+     */
+    _addLocalTrackAsUnmute(track) {
+        const allPromises = [ super._addLocalTrackAsUnmute(track) ];
+
+        if (this.p2pJingleSession) {
+            allPromises.push(this.p2pJingleSession.addTrackAsUnmute(track));
+        }
+
+        return Promise.all(allPromises);
+    }
+
+    /**
+     * Attaches local tracks back to the JVB connection.
+     * @private
+     */
+    _attachLocalTracksToJvbSession() {
+        const localTracks = this.getLocalTracks();
+
+        logger.info(`Attaching ${localTracks} to JVB`);
+        this.jvbJingleSession.attachLocalTracks(localTracks).then(
+            () => {
+                logger.info(`Attach ${localTracks} to JVB success!`);
+            },
+            error => {
+                logger.error(
+                    `Attach ${localTracks} to JVB failed!`, error);
+            });
+    }
+
+    /**
+     * Adds remote tracks to the conference associated with the JVB session.
+     * @private
+     */
+    _addRemoteJVBTracks() {
+        this._addRemoteTracks('JVB', this.jvbJingleSession);
+    }
+
+    /**
+     * Adds remote tracks to the conference associated with the P2P session.
+     * @private
+     */
+    _addRemoteP2PTracks() {
+        this._addRemoteTracks('P2P', this.p2pJingleSession);
+    }
+
+    /**
+     * Generates fake "remote track added" events for given Jingle session.
+     * @param {string} logName the session's nickname which will appear in log
+     * messages.
+     * @param {JingleSessionPC} jingleSession the session for which remote
+     * tracks will be added.
+     * @private
+     */
+    _addRemoteTracks(logName, jingleSession) {
+        if (!jingleSession) {
+            logger.info(
+                `Not adding remote ${logName} tracks - no session yet`);
+
+            return;
+        }
+        const remoteTracks = jingleSession.peerconnection.getRemoteTracks();
+
+        for (const track of remoteTracks) {
+            logger.info(`Adding remote ${logName} track: ${track}`);
+            this.rtc.eventEmitter.emit(RTCEvents.REMOTE_TRACK_ADDED, track);
+        }
+    }
+
+    /**
+     * @inheritDoc
+     * @override
+     */
+    _doReplaceTrack(oldTrack, newTrack) {
+        const allPromises = [ super._doReplaceTrack(oldTrack, newTrack) ];
+
+        if (this.p2pJingleSession) {
+            allPromises.push(
+                this.p2pJingleSession.replaceTrack(oldTrack, newTrack));
+        }
+
+        return Promise.all(allPromises);
+    }
+
+    /* eslint-disable prefer-rest-params */
```

Comment:
but they are inconvenient for this particular case, I would have to merge arrays.